automate-collections-updates/automate_collections_updates.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+"""Update requirements.yml with latest collection versions."""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import yaml
+import tarfile
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--vendored_collections",
+        type=str,
+        default=os.environ.get("VENDORED_COLLECTIONS", "vendored_collections.yml"),
+        help="Path/filename for vendored_collections.yml",
+    )
+    parser.add_argument(
+        "--repository",
+        type=str,
+        default=os.environ.get("REPOSITORY", "rhel-system-roles"),
+        help="The repository to work on. Either rhel-system-roles or linux-system-roles",
+    )
+    args = parser.parse_args()
+
+    hsh = yaml.safe_load(open(args.vendored_collections))
+    collection_tarballs = {}
+
+    repo = args.repository
+    if repo == "rhel-system-roles":
+        repo_url = "https://src.osci.redhat.com:443/rpms/rhel-system-roles.git"
+        branch = "rhel-9.0.0"
+        pkg_command = "rhpkg"
+    elif repo == "linux-system-roles":
+        repo_url = "https://src.fedoraproject.org/rpms/linux-system-roles.git"
+        branch = "rawhide"
+        pkg_command = "fedpkg"
+    else:
+        sys.exit(
+            "repository must be set to either rhel-system-roles or linux-system-roles"
+        )
+
+    for coll in hsh["collections"]:
+        coll_install_dir = tempfile.mkdtemp()
+        coll_name = coll["name"]
+        coll_content_values = set()
+        for values in list(coll["content"].values()):
+            for value in values:
+                coll_content_values.add(value)
+        coll_dir = os.path.join(
+            coll_install_dir,
+            "ansible_collections",
+            coll_name.split(".")[0],
+            coll_name.split(".")[1],
+        )
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
```

Comment:
Can you use `download` instead of `install`, to directly download a .tar.gz file?  Or is it that the logic that figures out what is the latest release only works with `install` and not `download`?

Reply:
`ansible-galaxy collection` cannot download tar.gz only. The best it can do is the following
`-n, --no-deps         Don't download collections listed as dependencies.`
So the script needs to install the collection and then tar it. Luckily, the format of the collection installed through `ansible-galaxy` is the same as in the tar downloaded from AH via UI.

Reply:
?  I'm using this logic in another script and it works fine:
`ansible-galaxy collection download -vv -p . ansible.posix`
e.g.
```
Starting collection download process to '/home/rmeggins/rh-dist-git/tests/rhel-system-roles/Library/basic/ansible.posix'
Downloading https://galaxy.ansible.com/download/ansible-posix-1.3.0.tar.gz to /home/rmeggins/.ansible/tmp/ansible-local-36821042yavan3/tmpz01gfpll/ansible-posix-1.3.0-tmwjrmic
Downloading collection 'ansible.posix:1.3.0' to '/home/rmeggins/rh-dist-git/tests/rhel-system-roles/Library/basic/ansible.posix'
Collection 'ansible.posix:1.3.0' was downloaded successfully
Writing requirements.yml file of downloaded collections to '/home/rmeggins/rh-dist-git/tests/rhel-system-roles/Library/basic/ansible.posix/requirements.yml'
```
this creates the file `ansible-posix-1.3.0.tar.gz`

Reply:
Yuy, it does work with `ansible-galaxy` from `ansible-core`, I'll use it here.